cake doesn't compile on kernel 4.12

I can't copy and paste the error but it says the error is from sch_cake.c in function cake_change
line 1743 error is too few arguments for function nla_parse_nested

full error below

make -C /lib/modules/4.12.1-2-ARCH/build SUBDIRS=/root/sch_cake modules LDFLAGS_MODULE="--build-id=0x14cbb5e76785b488c290f27c9f9523a9a71f6196" CFLAGS_MODULE="-DCAKE_VERSION=\\\"14cbb5e76785b488c290f27c9f9523a9a71f6196\\\""
make[1]: Entering directory '/usr/lib/modules/4.12.1-2-ARCH/build'
  CC [M]  /root/sch_cake/sch_cake.o
/root/sch_cake/sch_cake.c: In function ‘cake_change’:
/root/sch_cake/sch_cake.c:1743:8: error: too few arguments to function ‘nla_parse_nested’
  err = nla_parse_nested(tb, TCA_CAKE_MAX, opt, cake_policy);
        ^~~~~~~~~~~~~~~~
In file included from /root/sch_cake/sch_cake.c:55:0:
./include/net/netlink.h:754:19: note: declared here
 static inline int nla_parse_nested(struct nlattr *tb[], int maxtype,
                   ^~~~~~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:309: /root/sch_cake/sch_cake.o] Error 1
make[1]: *** [Makefile:1512: _module_/root/sch_cake] Error 2
make[1]: Leaving directory '/usr/lib/modules/4.12.1-2-ARCH/build'
make: *** [Makefile:8: default] Error 2

--- a/sch_cake.c
+++ b/sch_cake.c
@@ -1684,8 +1684,12 @@ static int cake_change(struct Qdisc *sch, struct nlattr *opt)
 
 	if (!opt)
 		return -EINVAL;
-
+#if KERNEL_VERSION(4,12,0) > LINUX_VERSION_CODE
 	err = nla_parse_nested(tb, TCA_CAKE_MAX, opt, cake_policy);
+#else
+	err = nla_parse_nested(tb, TCA_CAKE_MAX, opt, cake_policy, NULL);
+#endif
+	
 	if (err < 0)
 		return err;
 

